nrpy/infrastructures/BHaH/numerical_grids_and_timestep.py

```diff
@@ -153,16 +154,20 @@ def register_CFunction_numerical_grid_params_Nxx_dxx_xx(
 for (int j = 0; j < params->Nxx_plus_2NGHOSTS1; j++) xx[1][j] = params->xxmin1 + ((REAL)(j - NGHOSTS) + (1.0 / 2.0)) * params->dxx1;
 for (int j = 0; j < params->Nxx_plus_2NGHOSTS2; j++) xx[2][j] = params->xxmin2 + ((REAL)(j - NGHOSTS) + (1.0 / 2.0)) * params->dxx2;
 """
-    cfc.register_CFunction(
```

Comment:
Again, this should be reverted; the check for re-registering CFunctions is healthy as it prevents unnecessary work and should not be overridden. What should happen is the list of CoordSystems in the example code driver (nrpyelliptic) needs to be unique-ified.

Reply:
All of the function calls note list of coordinate systems, none of the related comments note that this should be unique.  Also, enforcing a set rather than list is counter intuitive on the user side where they attempting to construct a multipatch.

While I don't disagree with your justification, I think enforcing a set should not be on the application side and, rather, where it is needed.  An example could be in `CurviBoundaryConditions_register_C_functions`:

```
for CoordSystem in set(list_of_CoordSystems):
        # Register C function to set up the boundary condition struct.
        register_CFunction_bcstruct_set_up(CoordSystem=CoordSystem, fp_type=fp_type)

        # Register C function to apply boundary conditions to both pure outer and inner boundary points.
        register_CFunction_apply_bcs_outerradiation_and_inner(
            CoordSystem=CoordSystem,
            radiation_BC_fd_order=radiation_BC_fd_order,
            fp_type=fp_type,
        )
```

Reply:
Fair point. I would argue that the correct course of action is to replace list_of_CoordSystems with set_of_CoordSystems. I don't know why I'd need this to be a list.